### DISCUSSION TITLE
How do I read multiple branches with jagged structure through ```uproot.iterate()```

### DISCUSSION THREAD

**Discussion by MoAly98** [scikit-hep/uproot5#430] [Q&A]
Assume I have 3 branches in an input file called ```file``` in a tree ```tree```:
```branch1: type = AsJagged(AsDtype('>f4'))```
```branch2: type = AsJagged(AsDtype('>f4'))```
```branch3: type = AsDtype('>f4')```
My aim is to read in these 3 branches using uproot and save them into a dataframe. If I run 
```
for batch in uproot.iterate(file:tree, filter_name=[branch1, branch2, branch3], library='pd'):
      print(batch)
```

I run into the following error from the iterate method call:
```
  File "/afs/cern.ch/user/m/maly/thbb_env/lib/python3.7/site-packages/uproot/behaviors/TBranch.py", line 226, in iterate
    arrays = library.global_index(item, global_offset)
  File "/afs/cern.ch/user/m/maly/thbb_env/lib/python3.7/site-packages/uproot/interpretation/library.py", line 1084, in global_index
    index = arrays.index.values  # pandas<0.24.0
AttributeError: 'builtin_function_or_method' object has no attribute 'values'

```

This problem is not present when I read the same 3 branches with 
```
uproot.open(file:tree).arrays(filter_name=[branch1, branch2, branch3], library = 'pd' )
```
, which yields a tuple of dataframes with length = number of jagged arrays being read as expected (which is 2 here). 
I can't see anything in the ```uproot``` documentation which prohibiits reading multiple jagged arrays into pandas dataframes via the ```iterate``` method, so I'm not sure what is causing the above error. 

I am using version ```4.0.0``` of ```uproot```.

**Comment by jpivarski** [comment]
I moved this Discussion to Issue #433 because it's a bug-report. (I also moved my answer there.)
